### ISSUE TITLE
Problem creating token

### ISSUE THREAD

**Issue by Daniel290499** [bshaffer/oauth2-demo-php#58]
I downloaded your OAuth 2 API server a little while ago.
When I try to your user client and server, I get to the first page, click the authorize button.
I get to the next page, click the “Yes…” button.
I then get to the next page and click the “make a token request” button, and this is where things go wrong.
On the next page, I get the following error: Catchable fatal error: Argument 2 passed to GuzzleHttp\Client::post() must be of the type array, null given, called in F:\Work\Programming\PHP\Front End\oauth_server\src\OAuth2Demo\Client\Controllers\RequestToken.php on line 56 and defined in F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\Client.php on line 148
 
I traced the problem to a line of code in the request token script: $response = $http->post($endpoint, null, $query, $config['http_options'])->send();
I also printed out the variable to see if I could figure out what is happening, and the values for the above call are
string(41) "http://www.oauth_server.loc/lockdin/token"
 
array(5) { ["grant_type"]=> string(18) "authorization_code" ["code"]=> string(40) "c2d57abd35492642016e4a14b0252578bb95a173" ["client_id"]=> string(7) "demoapp" ["client_secret"]=> string(8) "demopass" ["redirect_uri"]=> string(51) "http://www.oauth_server.loc/client/receive_authcode" }
 
array(1) { ["exceptions"]=> bool(false) }
 
After trying your client, I tried to create my own client, and get to the same point again.
I’ve tried all sorts of things, but as soon as I create a post request to the /lockdin/token page, I get the following error message: Fatal error: Uncaught exception 'GuzzleHttp\Exception\ClientException' with message 'Client error response [url] http://www.oauth_server.loc/lockdin/token [status code] 400 [reason phrase] Bad Request' in F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\Exception\RequestException.php:89 Stack trace: #0 F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\Subscriber\HttpError.php(33): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Message\Request), Object(GuzzleHttp\Message\Response)) #1 F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\Event\Emitter.php(109): GuzzleHttp\Subscriber\HttpError->onComplete(Object(GuzzleHttp\Event\CompleteEvent), 'complete') #2 F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\RequestFsm.php(91): GuzzleHttp\Event\Emitter->emit('complete', Object(GuzzleHttp\Event\CompleteEvent)) #3 F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\RequestFsm.php(132): Guzzl in F:\Work\Programming\PHP\Back End\lib\composer\vendor\guzzlehttp\guzzle\src\Exception\RequestException.php on line 89
Basically, bad request.
This is the most basic request that I tried: $o_request = $o_client -> post('http://www.oauth_server.loc/lockdin/token');
 
Any chance you can help me here?
Do you know how I can fix this?

Regards
Daniel

**Comment by bshaffer** [comment]
You are (somehow) using the wrong version of Guzzle. The class you are referencing, `GuzzleHttp\Client`, is part of Guzzle5 and above, not the one used by this library. This library uses [`Guzzle\Client`](https://github.com/guzzle/guzzle3/blob/master/src/Guzzle/Http/Client.php), which is in Guzzle3.

Please try using the correct version of Guzzle and see if this fixes your issue.

**Comment by Daniel290499** [comment]
Aaaah winner!
You are correct sir.
Thanks for your help.

Is it possible to update this to make it work with the latest version of Guzzle?

**Comment by bshaffer** [comment]
It sure is possible! But it is not very high on my personal to-do list, so realistically it would have to be done by someone in the community.

**Comment by Daniel290499** [comment]
Cool. I'll give this a go towards the end of the week.
I'm still new to all this, so if anyone has some good documents or tutorials that they know of, that will be great! :)
